### PR TITLE
Move in-app logic into adapater (bug 1113801)

### PR DESCRIPTION
**Pull request by kumar303**
@muffinresearch r?

~~I still need to document the adapter methods but~~ this is my idea so far at how to separate out the in-app logic from the Firefox Marketplace logic. Next steps in future patches would be:

- implement a Firefox Marketplace adapter
- re-org the fxpay module so we can build a separate version without any in-app stuff

### CODE REVIEW

File: lib/fxpay/adapter.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  var API = require('fxpay/api').API;
+  var products = require('fxpay/products');
+  var receipts = require('fxpay/receipts');
+  var settings = require('fxpay/settings');
+  var utils = require('fxpay/utils');
+
+
+  function FxInappAdapter() {
+    //
+    // Adapter for Firefox Marketplace in-app products.
```
Comment (muffinresearch) [inline lib/fxpay/adapter.js]: Some of this comment seems confusing to me.This is the in-app adapter isn't it - as opposed to the one that hasn't been written for the marketplace specifics? The comment seems to mention the latter which muddies things for me - though granted I'm easily confused :)

File: lib/fxpay/index.js [hunk 2]
```diff
@@ -76,23 +58,59 @@
...
     opt.pollIntervalMs = opt.pollIntervalMs || undefined;
-    var productInfo = {};
+
+    var partialProdInfo = {productId: productId};
+    var log = settings.log;
+    var paymentWindow;
 
     if (!onPurchase) {
-      onPurchase = function _onPurchase(err, info) {
+      onPurchase = function _onPurchase(err, returnedProdInfo) {
```
Comment (muffinresearch) [inline lib/fxpay/index.js]: Good call on the naming changes. :+1:

File: lib/fxpay/adapter.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+  FxInappAdapter.prototype.startTransaction = function(opt, callback) {
+    //
+    // Start a transaction.
+    //
+    // The `opt` object contains the following parameters:
+    //
+    // - productId: the ID of the product purchased.
+    //
+    // When finished, execute callback(error, transactionData).
```
Comment (muffinresearch) [inline lib/fxpay/adapter.js]: nice docs - thinking we could look at generating some code docs with these kind of comments. E.g. yuidoc or similar.

**Comment by muffinresearch** [comment]
r+wc

File: lib/fxpay/adapter.js (hunk 1 above)
Comment (kumar303) [inline lib/fxpay/adapter.js]: Yeah, good point. I reworded it.

**Pull request merged**